pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Producer.java

```diff
@@ -107,10 +107,19 @@
      */
     TypedMessageBuilder<T> newMessage();
 
+    /**
+     * Create a new message builder with schema specified explicitly,
+     *
+     * @return a typed message builder that can be used to construct the message to be sent through this producer
+     * @since 2.5.0
+     * @see #newMessage()
+     */
+    TypedMessageBuilder<T> newMessage(Schema<T> schema);
```

Comment:
it seems to me that this method is a bit confusing. T is the original type, no? shouldn't it be a different type?

```
<V> TypedMessageBuilder<V> newMessage(Schema<V> schema);
```

Reply:
@sijie yeah, you are right. I would change the signature in the following PR to make this one easier to review. It would break some more inner method of implementation class.

Reply:
I'll leave a TODO, or just hidden this interface for still not complete

Reply:
I have hidden this interface and left a TODO.